platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ConfiguratorImpl.java

```diff
@@ -0,0 +1,425 @@
+            }
+        }
+
+        return configReport;
+    }
+
+    /**
+     * Starts the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String startBundle(String bundleSymName) {
+        return registerHandler(BundleOperation.forStart(bundleSymName, getBundleContext()));
+    }
+
+    /**
+     * Stops the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String stopBundle(String bundleSymName) {
+        return registerHandler(BundleOperation.forStop(bundleSymName, getBundleContext()));
+    }
+
+    /**
+     * Determines if the bundle with the given name is started.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return true if started; else, false
+     */
+    public boolean isBundleStarted(String bundleSymName) {
+        try {
+            return BundleOperation.forStart(bundleSymName, getBundleContext())
+                    .readState();
+        } catch (ConfiguratorException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Installs and starts the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String startFeature(String featureName) {
+        return registerHandler(FeatureOperation.forStart(featureName, getBundleContext()));
+    }
+
+    /**
+     * Stops the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String stopFeature(String featureName) {
+        return registerHandler(FeatureOperation.forStop(featureName, getBundleContext()));
+    }
+
+    /**
+     * Determines if the feature with the given name is started.
+     *
+     * @param featureName the name of the feature
+     * @return true if started; else, false
+     */
+    public boolean isFeatureStarted(String featureName) {
+        return FeatureOperation.forStart(featureName, getBundleContext())
+                .readState();
+    }
+
+    /**
+     * Creates a property file in the system with the given set of new key:value pairs.
+     *
+     * @param propFile   the property file to create
+     * @param properties the set of key:value pairs to save to the property file
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String createPropertyFile(Path propFile, Map<String, String> properties) {
```

Comment:
When creating the file, do we check that it is within the valid locations constrained to just our system. In other words, make sure that we can't write files outside of DDF.